yt_dlp/extractor/smotrettv.py

```diff
@@ -0,0 +1,139 @@
+import urllib.parse
+
+from .common import InfoExtractor
+from ..utils import (
+    base_url,
+    clean_html,
+    extract_attributes,
+    get_domain,
+    get_element_by_class,
+    traverse_obj,
+    url_or_none,
+)
+
+
+class SmotretTVIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?smotret\.tv/(?P<id>[^/#?]+)'
+
+    _TESTS = [{
+        # direct streams
+        'url': 'https://smotret.tv/rossiya-1',
+        'info_dict': {
+            'id': 'rossiya-1',
+            'ext': 'mp4',
+            'title': 're:Россия 1 — смотреть онлайн прямой эфир [0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}$',
+            'description': str,
+            'thumbnail': 'https://smotret.tv/images/rossiya-1.jpg',
+            'live_status': 'is_live',
+            'subtitles': dict,
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }, {
+        # RutubeEmbed in iframe
+        'url': 'https://smotret.tv/otr',
+        'info_dict': {
+            'id': 'faa934385b83f9e8a92f5484defae5fa',  # TODO: use "otr"
```

Comment:
TODO: About `self.url_result()`, how do I preserve the id?

Reply:
No way for now, see https://github.com/yt-dlp/yt-dlp/pull/10831#discussion_r1729607085.